### ISSUE TITLE
Release 0.2.0

### ISSUE THREAD

**Issue by roblarsen** [h5bp/create-html5-boilerplate#35]
I'm happy to release early and often with this as we add more features. I'd like to release 0.2.0. This issue will track those activities.

**Comment by vltansky** [comment]
What left for 0.2 release?

**Comment by roblarsen** [comment]
Updating the changelog, I think. @coliff anything else?

**Comment by vltansky** [comment]
Changelog:
- prompt: remove jQuery in < v8 https://github.com/h5bp/create-html5-boilerplate/pull/12 https://github.com/h5bp/create-html5-boilerplate/pull/24
- readme fixes
- lang select https://github.com/h5bp/create-html5-boilerplate/pull/19
- check if the folder is empty https://github.com/h5bp/create-html5-boilerplate/pull/22
- update dependencies

@coliff @roblarsen did I miss something?

**Comment by roblarsen** [comment]
Looks good to me. Let's release this one. I'll make a PR for the changelog this afternoon

**Comment by roblarsen** [comment]
I'm going to push out an RC release this afternoon.

**Comment by roblarsen** [comment]
I'm the worst! I will publish this weekend.

**Comment by roblarsen** [comment]
@vltansky I published the RC1 package yesterday and tested. It works as expected- except it doesn't exit back to the command line. You have to kill the process to get the command line back to being active. That's on Git Bash at least. I didn't test with Powershell and can't test on a Mac. Other than that it's ready to go.

**Comment by vltansky** [comment]
@roblarsen till Monday I don't have access to computer. So it's up to you to decide if it's smth critical. Will check it when will back home

**Comment by roblarsen** [comment]
Hi @vltansky Any idea on what needs to happen here?

**Comment by vltansky** [comment]
![image](https://user-images.githubusercontent.com/5851280/93715187-740adf80-fb70-11ea-9b94-ca74acf50997.png)

![image](https://user-images.githubusercontent.com/5851280/93715192-8d139080-fb70-11ea-9cfd-208fea4d0a49.png)

Works fine for me both windows cmd and git bash (on windows)

**Comment by roblarsen** [comment]
@vltansky are you testing it from npm, not the linked local version?

**Comment by vltansky** [comment]
@roblarsen you're right I tested with the local version.
But now I deleted it and it still works fine:
![image](https://user-images.githubusercontent.com/5851280/93715681-728ee680-fb73-11ea-87cf-425de89d0f30.png)

**Comment by roblarsen** [comment]
I just tested again and got the same result. I'll run through the specs of the system at some point today

![image](https://user-images.githubusercontent.com/361421/93718309-43fc1600-fb49-11ea-90e0-68e8b3fa04bf.png)

**Comment by vltansky** [comment]
@roblarsen try with the previous versions to test if it's something that we changed since then:
`npx create-html5-boilerplate@0.0.1`

**Comment by roblarsen** [comment]
@vltansky using v0.1.0 works

That system (which is older) has node 10.16.3 and npm 6.9.0 I'm going to upgrade both to see if that does something

**Comment by roblarsen** [comment]
It also happens on a new machine- node v12.13.1 npm v6.12.1

**Comment by vltansky** [comment]
Still can't reproduce... (not in CMD nor in Git Bash)
![image](https://user-images.githubusercontent.com/5851280/95717821-36343f00-0c76-11eb-84bd-54858c3b057e.png)

How can we proceed?

**Comment by roblarsen** [comment]
@vltansky  I'll try to take a look to see what's happening on my machines. Once I do that we can actually just release anyway- people are using this now and no one has complained, so it's not a deal-breaker.

**Comment by roblarsen** [comment]
The problem is with [spinner.succeed()](https://github.com/h5bp/create-html5-boilerplate/blob/master/lib/cli.js#L44) We hit that line, it executes, printing the message to the console, but on the machine I'm on right now, it never turns the check mark green. So... it's "succeeding" but not completely. I am currently manually killing the process just so I don't have to type ctrl+c a million times while debugging, but something somewhere is causing an issue. I'll continue to debug, but it's a clearly reproduceable issue on my end, so we'll have to fix it before we can release.

**Comment by roblarsen** [comment]
I just realized I can't navigate the list of languages. I can hit <enter> an select English but I can't select anything else. I'm calling it a night

**Comment by vltansky** [comment]
I just merged PR that may fix that. Give a try on the master.

**Comment by roblarsen** [comment]
That change didn't do anything. I ended up just adding a manual process.exit() in the release branch. 

The language list still doesn't scroll in Git Bash. It does works in powershell, but not in Git bash, so the language select feature is  basically unusable. I'll continue to plug away at it.

**Comment by vltansky** [comment]
What is the version of your Git Bash?

**Comment by roblarsen** [comment]
3.0.6

**Comment by roblarsen** [comment]
As an aside I changed the workflow a little bit. We need to use branches. Pushing directly to master dirties the history.

**Comment by vltansky** [comment]
Can you give a try on this branch: https://github.com/vltansky/create-html5-boilerplate
I changed `prompts` to `inquirer`.

**Comment by roblarsen** [comment]
Cool. I'll take a look. Do you understand why the coveralls action is failing? Based on the error it should never have worked since the file it's looking for isn't in the repo.

**Comment by roblarsen** [comment]
When is it generated? 

At this point I don't care about anything other than getting the actual app to work. It fails in 2 of 3 systems I have access to. Everything else is just nice to have at this point.

**Comment by vltansky** [comment]
In the coveralls workflow. I guess exiting the process is exiting tests.
Try the branch with inquirer, I think the main problem is prompts autocomplete in git bash. If it won't work I'll try to downgrade git bash on my PC in order to reproduce the bug.

**Comment by roblarsen** [comment]
Ah, got it.

I just upgraded Git bash on one of the machines that show errors- the new code exits fine (👍 ) and allows you to scroll down through the list of languages, but doesn't highlight them and doesn't scroll past the first screenful of languages

**Comment by vltansky** [comment]
Didn't understand, can you attach screenshots?

**Comment by vltansky** [comment]
Is inquirer now works on old git bash but without filtering?
Is it working fine on new git bash?

**Comment by roblarsen** [comment]
video:
https://youtu.be/-1QaFYahrBc

**Comment by roblarsen** [comment]
The filtering doesn't work at all.

**Comment by vltansky** [comment]
So the bug is in git bash 3.0.6, right? New git bash works fine ?
I'm not sure if we can fix it at all, maybe we should open issue to inquirer/prompts?
Anyway I'll downgrade my git bash and will try to find some workaround.

**Comment by roblarsen** [comment]
No, that screen shot is with 3.4 I just upgraded

**Comment by roblarsen** [comment]
![image](https://user-images.githubusercontent.com/361421/96741778-34622e00-1390-11eb-982a-115164483ef1.png)

**Comment by vltansky** [comment]
https://github.com/ionic-team/ionic-cli/issues/2807
https://github.com/SBoudrias/Inquirer.js/issues/570
https://github.com/angular/angular-cli/issues/18163

Looks like it's a common bug with git bash. Try running git bash from vscode.

**Comment by roblarsen** [comment]
What I don't understand is why it works on some machines and not others.

**Comment by roblarsen** [comment]
Okay, let's release 0.2.0 (we're not at 1.0.0 so things can be a little wonky, especially in a case like this where it's really weird and outside of our control.)

What I will do is write up an issue to _improve_ the way this works across different console apps. I am also interested in reaching out to some folks to get a better understanding of why it works sometimes and not others. This is all new coding for me so I'll have to do a crash course.

**Comment by roblarsen** [comment]
The coveralls action is still failing

**Comment by vltansky** [comment]
> The coveralls action is still failing

You didnt remove `process.exit()` see line 63

Also try running this in Git Bash:
`node --eval "console.log({stdin: process.stdin.isTTY, stdout: process.stdout.isTTY})"`

**Comment by roblarsen** [comment]
Ah, good catch.

**Comment by vltansky** [comment]
Probably you will get false from `node --eval "console.log({stdin: process.stdin.isTTY, stdout: process.stdout.isTTY})"`
Which windows version do you have?
The good news that we can detect non-interactive consoles (https://www.npmjs.com/package/inquirer#use-in-non-interactive-environments), so we can disable prompts there.

https://github.com/nodejs/node/issues/3006

**Comment by roblarsen** [comment]
On both machines I get:

{ stdin: true, stdout: true }

**Comment by roblarsen** [comment]
Win 10 on both machines

**Comment by vltansky** [comment]
Then I don't know...
Looks the same issue like in Angular CLI:
https://github.com/angular/angular-cli/issues/18163#issuecomment-654795403_

**Comment by roblarsen** [comment]
Yeah, super weird. It's fine for now. I'll open a detailed issue and post it to twitter and write it up on my blog. Let me fix the coveralls issue and then I'll release 0.2.0

**Comment by vltansky** [comment]
We can release with semantic-release if you're interested #73

**Comment by roblarsen** [comment]
(Done, I'm writing it up today.)
